homeassistant/components/sensor/tof.py

```diff
@@ -0,0 +1,129 @@
+DEFAULT_I2C_BUS = 1
+DEFAULT_XSHUT = 16
+DEFAULT_RANGE = 2
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME,
+                 default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_I2C_ADDRESS,
+                 default=DEFAULT_I2C_ADDRESS): vol.Coerce(int),
+    vol.Optional(CONF_I2C_BUS,
+                 default=DEFAULT_I2C_BUS): vol.Coerce(int),
+    vol.Optional(CONF_XSHUT,
+                 default=DEFAULT_XSHUT): cv.positive_int,
+})
+
+
+def init_tof(xshut, level):
+    """XSHUT port LOW resets the device."""
+    rpi_gpio.setup_output(xshut)
+    rpi_gpio.write_output(xshut, level)
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_entities,
+                               discovery_info=None):
+    """Reset and initialize the VL53L1X ToF Sensor from STMicroelectronics."""
+    from VL53L1X2 import VL53L1X  # pylint: disable=import-error
+
+    name = config.get(CONF_NAME)
+    bus_number = config.get(CONF_I2C_BUS)
+    i2c_address = config.get(CONF_I2C_ADDRESS)
+    unit = LENGTH_MILLIMETERS
+    xshut = config.get(CONF_XSHUT)
+
+    await hass.async_add_executor_job(
+        init_tof, xshut, 0
+    )
+    await asyncio.sleep(0.01)
+    await hass.async_add_executor_job(
+        init_tof, xshut, 1
+    )
+    await asyncio.sleep(0.01)
+
+    sensor = await hass.async_add_executor_job(
+        partial(VL53L1X, bus_number)
+    )
+
+    sensor.open()
+
+    dev = [VL53L1XSensor(sensor, name, unit, i2c_address)]
+
+    async_add_entities(dev, True)
+
+
+class VL53L1XSensor(Entity):
+    """Implementation of VL53L1X sensor."""
+
+    def __init__(self, vl53l1x_sensor, name, unit, i2c_address):
+        """Initialize the sensor."""
+        self._name = name
+        self._unit_of_measurement = unit
+        self.vl53l1x_sensor = vl53l1x_sensor
+        self.i2c_address = i2c_address
+        self._state = None
+        self.vl53l1x_sensor.add_sensor(
+            self.i2c_address,
+            self.i2c_address
+        )
+        _LOGGER.info(
+            "Setup of VL53L1X sensor at %s is complete",
+            self.i2c_address
+        )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self) -> int:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    def measure(self):
+        """Get the latest measurement from VL53L1X."""
+        self.vl53l1x_sensor.start_ranging(self.i2c_address, DEFAULT_RANGE)
+        self.vl53l1x_sensor.update(self.i2c_address)
+        self.vl53l1x_sensor.stop_ranging(self.i2c_address)
+
+    async def async_update(self):
+        """Get the latest measurement and update state."""
+        await self.hass.async_add_executor_job(
```

Comment:
Remove this and call the methods now in `measure` directly here.

Reply:
I assume these calls do I/O and this was the reason we move it to async before.